### PR TITLE
RHDEVDOCS-3614 ODC Release Notes, New features for 4.10

### PR DESCRIPTION
**Pull request by rishumehra**
* Aligned team: Dev Tools
* For branches: enterprise-4.10 release notes
* Jira: [RHDEVDOCS-3614](https://issues.redhat.com/browse/RHDEVDOCS-3614)
* Direct link to doc preview: [Preview](https://deploy-preview-42818--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-web-console-developer-perspective)
* SME review: @invincibleJai @jerolimov 
* QE review: @sanketpathak @karthikjeeyar 
* Peer review: @rolfedh

### CODE REVIEW

File: release_notes/ocp-4-10-release-notes.adoc [hunk 1]
```diff
@@ -175,6 +175,15 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* You can now specify a name to a Service Binding connector in the *Topology* view while making a binding connection.
+* You can check if a pipeline is created and not started using the status decorator of the pod in the *Topology* view.
+* This enhancement gives you a better user experience when an application export is in progress a dialog box window or an alert will appear to view the logs, you can also cancel or restart the exporting process from this dialog box window.
+* You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.
```
Comment (jerolimov) [inline release_notes/ocp-4-10-release-notes.adoc:183]: Is this about namespaced helm chart repos? Then I would suggest to call the exact resource at the end:

```suggestion
* You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *ProjectHelmChartRepository*.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 2]
```diff
@@ -175,6 +175,15 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* You can now specify a name to a Service Binding connector in the *Topology* view while making a binding connection.
```
Comment (rolfedh) [inline release_notes/ocp-4-10-release-notes.adoc:180]:
```suggestion
* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
```
Lower case "service binding" unless you're talking about the "Service Binding Operator" proper noun.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 3]
```diff
@@ -175,6 +175,15 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
...
+* You can check if a pipeline is created and not started using the status decorator of the pod in the *Topology* view.
+* This enhancement gives you a better user experience when an application export is in progress a dialog box window or an alert will appear to view the logs, you can also cancel or restart the exporting process from this dialog box window.
```
Comment (rolfedh) [inline release_notes/ocp-4-10-release-notes.adoc:182]:
```suggestion
* This enhancement opens a dialog or alert that displays the export logs while an application export is in progress. You can use the dialog to cancel or restart the exporting process.
```

File: release_notes/ocp-4-10-release-notes.adoc (hunk 1 above)
Comment (rolfedh) [inline release_notes/ocp-4-10-release-notes.adoc:183]:
```suggestion
* With this update, you can add your new Helm Chart Repository to the Developer Catalog by creating a custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 4]
```diff
@@ -175,6 +175,15 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
...
+* You can now add your new Helm Chart Repository in the Developer Catalog by creating a new custom resource. Refer to the QuickStart guides in the Developer perspective to add a new *HelmChartRepo*.
+* You can now view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.
```
Comment (rolfedh) [inline release_notes/ocp-4-10-release-notes.adoc:184]:
```suggestion
* With this update, you can view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.
```

**Review by rolfedh — COMMENTED** [review]
@rishumehra  I hope these suggestions are helpful.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 2 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:180]: Thanks for pointing out the lower case. Will add this to the updates.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 5]
```diff
@@ -175,6 +175,15 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* You can now specify a name to a Service Binding connector in the *Topology* view while making a binding connection.
+* You can check if a pipeline is created and not started using the status decorator of the pod in the *Topology* view.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:181]: @rolfedh  Your rephrase seems fine in the suggestion. "The pod" is enough also.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 3 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:182]: Good suggestion.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 1 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:183]: Yes @jerolimov it is about namespaced helm chart repos. I'll update the text with the exact resource.

Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:183]: Done

File: release_notes/ocp-4-10-release-notes.adoc (hunk 4 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:184]: Done

**Review by rishumehra — COMMENTED** [review]
Thanks, @rolfedh @jerolimov for your reviews. I'll commit new updates soon.

**Comment by rishumehra** [comment]
> * Aligned team: Dev Tools
> * For branches: enterprise-4.10 release notes
> * Jira: [RHDEVDOCS-3614](https://issues.redhat.com/browse/RHDEVDOCS-3614)
> * Direct link to doc preview: TBD
> * SME review: @invincibleJai
> * QE review: @sanketpathak
> * Peer review: @rolfedh

**Review by rolfedh — APPROVED** [review]

**Review by christoph-jerolimov — COMMENTED** [review]
I don't verify if this contains all epics. But technically this lgtm :+1:

/lgtm

**Comment by jerolimov** [comment]
Hey, I compared this now with our [13 epics](https://issues.redhat.com/issues/?filter=12383856) we have delivered with 4.10. :)

First of all, I think this epics are already addressed well :+1:

* [x] [ODC-5947](https://issues.redhat.com/browse/ODC-5947) Support Service Binding in Topology view
* [x] [ODC-5506](https://issues.redhat.com/browse/ODC-5506) Devfile catalog integration
* [x] [ODC-5713](https://issues.redhat.com/browse/ODC-5713) Namespaced Helm Chart Registries
* [x] [ODC-6296](https://issues.redhat.com/browse/ODC-6296) Enhance Export App UX experience

But maybe we want to mention some of these epics here. I added on the second level the feature I would mention. But if you want or don't want pick one up, and the option to aggregate them are obviously up to you. Maybe we don't want mention everything...? 😄 

* [ ] https://issues.redhat.com/browse/ODC-6266 Improve DevExp for front end developers
  * [ ] There is a new option to override the npm start command when import an application from Git with s Node.js Builder Image.
  * [ ] We simplified the access to the target port field when importing an application.
  * [ ] After importing there is new toast notification which shows the exported route URL with an option to copy it also.
  * [x] The devfile catalog shows devfiles from the community devfile registry
* [ ] https://issues.redhat.com/browse/ODC-6303 Dev Console Usability Improvements
  * [ ] The add page has now the option to open the same Quick Add/Search modal similar to the topology
  * [ ] There are new user preferences to define the default security options for new Deployments
* [ ] https://issues.redhat.com/browse/ODC-6359 ODC/Serverless Usability Improvements
  * [ ] Improved visualization of Kamelet sink resource in topology
* https://issues.redhat.com/browse/ODC-6372 Add flows should allow user to pick from a list of pipelines
  * [ ] Custom pipelines are now also shown on the import page and could be used when import an application from source or container image.
* [ ] https://issues.redhat.com/browse/ODC-6361 ODC/Topology Usability Improvements
  * [ ] Users can now resize the sidebar in topology based on their needs.
  * [ ] The latest selection was also keeped open when switching between topology and other pages.
  * [ ] The topology 'Open URL' decorator on Deployments could now be customized with some annotations.

There are many topology-related improvements... Maybe it's enough to aggregate them?

I also agree that we don't have to add these two epics (IMHO!), as well as the automation epic. I can see that the dynamic plugin epic is already mentioned below your list.

* [x] ~~[ODC-6375](https://issues.redhat.com/browse/ODC-6375) Guide user to add webhook for pipeline templates~~
* [x] ~~[ODC-6377](https://issues.redhat.com/browse/ODC-6377) ODC/Pipelines TektonHub Improvements~~

**Comment by jerolimov** [comment]
Thanks Rishu for checking my list above and the clarification that these things should not be included in the release notes. Added some individual updates above why.

/lgtm

**Comment by Preeticp** [comment]
I recommend adding [ODC-6375](https://issues.redhat.com/browse/ODC-6375) Guide user to add webhook for pipeline templates
 [ODC-6377](https://issues.redhat.com/browse/ODC-6377) ODC/Pipelines TektonHub Improvements as these are important features from the end user POV for Pipelines.
 Maybe add three nested-bullets instead of this one: * With this update, you can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
 Something like -
 The creating pipelines workflow has now been enhanced:
-  You can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
- webhook related info
- Hub integration

**Comment by rishumehra** [comment]
> I recommend adding [ODC-6375](https://issues.redhat.com/browse/ODC-6375) Guide user to add webhook for pipeline templates and [ODC-6377](https://issues.redhat.com/browse/ODC-6377) ODC/Pipelines TektonHub Improvements, as these are important features from the end user POV for Pipelines. Maybe add three nested-bullets instead of this one: * With this update, you can see when the pipeline has been created but not started by looking at the status of the pod in the _Topology_ view. Something like - The creating pipelines workflow has now been enhanced:
> 
> * You can see when the pipeline has been created but not started by looking at the status of the pod in the _Topology_ view.
> * webhook related info
> * Hub integration

@Preeticp Thanks for the suggestion I'll add it in further edits.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 6]
```diff
@@ -175,6 +175,29 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
+* With this update, creating pipelines workflow has now been enhanced:
+** You can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
+** You can create an application using the *Import from Git* workflow, along with a template-generated pipeline by selecting the *Add pipeline* option. The webhook resources and URL are visible in the side panel of the selected resources.
+** You can opt out of the default Tekton Hub integration by setting the parameter `enable-devconsole-integration` to `false` in the `TektonConfig` custom resource. When you opt out, only the platform compatible cluster tasks for your namespace are listed in the *Pipeline builder* -> *Add tasks* view.
```
Comment (Preeticp) [inline release_notes/ocp-4-10-release-notes.adoc:184]: @karthikjeeyar can you please verify if this is technically correct?

Comment (karthikjeeyar) [inline release_notes/ocp-4-10-release-notes.adoc:184]: I Suggest the following changes, Please update the text as you see fit.

```suggestion
** Pipeline builder will contain the Tekton Hub tasks that are supported by the cluster's Architecture, all other unsupported tasks are excluded from the list.
** You can opt out of the default Tekton Hub integration by setting the parameter `enable-devconsole-integration` to `false` in the `TektonConfig` custom resource.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 7]
```diff
@@ -175,6 +175,29 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
+* With this update, creating pipelines workflow has now been enhanced:
+** You can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
+** You can create an application using the *Import from Git* workflow, along with a template-generated pipeline by selecting the *Add pipeline* option. The webhook resources and URL are visible in the side panel of the selected resources.
```
Comment (karthikjeeyar) [inline release_notes/ocp-4-10-release-notes.adoc:183]:
```suggestion
**You can now choose an user defined pipeline from a dropdown list while importing your application from import from Git/Docker pipeline workflow.
**Default webhooks are added for the pipelines that are created using Add flow and URL is visible in the side panel of the selected resources.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 8]
```diff
@@ -175,6 +175,14 @@ For more information, see xref:../updating/understanding-upgrade-channels-releas
 [id="ocp-4-10-web-console"]
 === Web console
 
+[id="ocp-4-10-web-console-developer-perspective"]
+==== Developer perspective
+* With this update, you can specify the name of a service binding connector in the *Topology* view while making a binding connection.
+* With this update, you can see when the pipeline has been created but not started by looking at the status of the pod in the *Topology* view.
+* This enhancement opens a dialog or alert that displays the export logs while an application export is in progress. You can use the dialog to cancel or restart the exporting process.
+* With this update, you can add your new Helm Chart Repository to the Developer Catalog by creating a custom resource. Refer to the quick start guides in the Developer perspective to add a new *ProjectHelmChartRepository*.
+* With this update, you can view and use devfiles samples from (link:http://registry.devfile.io/[http://registry.devfile.io/]). These samples are present in the Developer Catalog.
```
[the remark this answers is not in the document]
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:199]: Done

File: release_notes/ocp-4-10-release-notes.adoc (hunk 6 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:184]: Thanks for the suggestion @karthikjeeyar I'll incorporate these changes in the next edit.

File: release_notes/ocp-4-10-release-notes.adoc (hunk 7 above)
Comment (rishumehra) [reply release_notes/ocp-4-10-release-notes.adoc:183]: Thanks for the suggestion @karthikjeeyar

**Review by rishumehra — COMMENTED** [review]
@karthikjeeyar @Preeticp thanks for your reviews and comment. I'll try to rephrase the content as per the suggestions.

**Review by karthikjeeyar — APPROVED** [review]
LGTM

**Pull request merged**